### PR TITLE
add DOLT_BRANCH() function

### PR DESCRIPTION
**Pull request by jennifersp**

### CODE REVIEW

File: integration-tests/bats/sql-branch.bats [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+}
+
+@test "sql-branch: DOLT_BRANCH works" {
+    run dolt branch
+    [[ ! "$output" =~ "new_branch" ]] || false
+
+    run dolt sql -q "SELECT DOLT_BRANCH('new-branch')"
+    [ $status -eq 0 ]
+
+    # dolt sql -q "select dolt_branch() should not change the branch
```
Comment (timsehn) [inline integration-tests/bats/sql-branch.bats:28]: I think you can remove this comment because the behavior matches dolt branch on the CLI.

**Review by timsehn — APPROVED** [review]
Other than that Bats looks good.

File: integration-tests/bats/sql-branch.bats [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+@test "sql-branch: DOLT_BRANCH works" {
+    run dolt branch
+    [[ ! "$output" =~ "new_branch" ]] || false
+
+    run dolt sql -q "SELECT DOLT_BRANCH('new-branch')"
+    [ $status -eq 0 ]
+
+    # It changes the branch for that session which ends after the SQL
+    # statements are executed.
```
Comment (andrew-wm-arthur) [inline integration-tests/bats/sql-branch.bats:29]: I don't understand this comment. Are you saying that the SQL session for line 25 has its branch changed? 

`git branch abc` and `dolt branch abc` both create a new branch, but do not switch to it.

File: go/libraries/doltcore/sqle/dfunctions/dolt_branch.go [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+	case apr.Contains(cli.DeleteFlag):
+		//err = deleteBranches(ctx, dbData, apr, apr.Contains(cli.ForceFlag))
+		if err != nil {
+			return 1, errors.New("Deleting branches is not supported.")
+		}
+	case apr.Contains(cli.DeleteForceFlag):
+		//err = deleteBranches(ctx, dbData, apr, true)
+		if err != nil {
+			return 1, errors.New("Deleting branches is not supported.")
+		}
```
Comment (andrew-wm-arthur) [inline go/libraries/doltcore/sqle/dfunctions/dolt_branch.go:109]: I would add one more bats test asserting that these actions are prevented

**Review by andrew-wm-arthur — APPROVED** [review]
LGTM, couple comments

**Pull request merged**